src/convertToObject.js

```diff
@@ -6,7 +6,25 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const source = sourceString;
+  const splitedByASemicolon = source
+    .split(';')
+    .map((string) => string.trim())
+    .filter((string) => string.length > 0);
+  const arrOfArrsSplitedByAColon = [];
+
+  splitedByASemicolon.forEach((string) => {
+    arrOfArrsSplitedByAColon.push(string.split(':'));
+  });
+
+  const finishedArray = arrOfArrsSplitedByAColon.map((element) => {
+    return element.map((string) => string.trim());
+  });
+  const finishedObject = {};
+
+  finishedArray.map(([key, value]) => (finishedObject[key] = value));
```

Comment:
[CODE KNOWLEDGE] - The `map` method is intended to transform elements in an array and return a new array. Here, you are using it to iterate over `finishedArray` without utilizing its return value. This is a misuse of `map`. You should use `forEach` here since you are not using the return value of `map`.